src/System.Runtime.Extensions/src/System/IO/TextReader.cs

```diff
@@ -100,6 +101,27 @@ public virtual int Read(char[] buffer, int index, int count)
             return n;
         }
 
+        // Reads a span of characters. This method will read up to
+        // count characters from this TextReader into the
+        // span of characters Returns the actual number of characters read.
+        //
+        public virtual int Read(Span<char> destination)
+        {
+            char[] buffer = ArrayPool<char>.Shared.Rent(destination.Length);
+
+            try
+            {
+                int numRead = Read(buffer, 0, destination.Length);
+                if ((uint)numRead > destination.Length)
+                {
+                    throw new IOException(SR.IO_StreamTooLong);
+                }
+                new Span<char>(buffer, 0, numRead).CopyTo(destination);
+                return numRead;
+            }
+            finally { ArrayPool<char>.Shared.Return(buffer); }
```

Comment:
per style, can you please put braces on their own lines

Reply:
Did have a thought about that after I took that code from [here](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/IO/Stream.cs#L801)... will follow that now on..,